examples/random_map.rs

```diff
@@ -1,61 +1,61 @@
-use std::collections::HashSet;
-
 use bevy::{
-    core::Time,
     diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin},
     prelude::*,
-    render::camera::OrthographicCameraBundle,
-    window::WindowDescriptor,
 };
-use bevy_ecs_tilemap::prelude::*;
+use bevy_ecs_tilemap::{
+    map::{
+        Tilemap2dGridSize, Tilemap2dSize, Tilemap2dTextureSize, Tilemap2dTileSize, TilemapId,
+        TilemapTexture,
+    },
+    tiles::{Tile2dStorage, TilePos2d, TileTexture, TileVisible},
+    Tilemap2dPlugin, TilemapBundle,
+};
 use rand::{thread_rng, Rng};
 
 mod helpers;
 
-fn startup(mut commands: Commands, asset_server: Res<AssetServer>, mut map_query: MapQuery) {
+fn startup(mut commands: Commands, asset_server: Res<AssetServer>) {
     commands.spawn_bundle(OrthographicCameraBundle::new_2d());
 
-    let texture_handle = asset_server.load("tiles.png");
-
-    // Create map entity and component:
-    let map_entity = commands.spawn().id();
-    let mut map = Map::new(0u16, map_entity);
+    let texture_handle: Handle<Image> = asset_server.load("tiles.png");
 
-    let layer_settings = LayerSettings::new(
-        MapSize(10, 10),
-        ChunkSize(64, 64),
-        TileSize(16.0, 16.0),
-        TextureSize(96.0, 16.0),
-    );
+    let tilemap_size = Tilemap2dSize { x: 640, y: 640 };
+    let mut tile_storage = Tile2dStorage::empty(tilemap_size);
+    let tilemap_entity = commands.spawn().id();
 
-    let center = layer_settings.get_pixel_center();
-
-    // Chunk sizes of 64x64 seem optimal for meshing updates.
-    let (mut layer_builder, layer_entity) =
-        LayerBuilder::<TileBundle>::new(&mut commands, layer_settings, 0u16, 0u16);
-    map.add_layer(&mut commands, 0u16, layer_entity);
-
-    layer_builder.for_each_tiles_mut(|tile_entity, tile_data| {
-        // True here refers to tile visibility.
-        *tile_data = Some(TileBundle::default());
-        // Tile entity might not exist at this point so you'll need to create it.
-        if tile_entity.is_none() {
-            *tile_entity = Some(commands.spawn().id());
+    for x in 0..640u32 {
+        for y in 0..640u32 {
+            let tile_pos = TilePos2d { x, y };
+            let tile_entity = commands
+                .spawn()
+                .insert(tile_pos)
+                .insert(TileTexture(0))
+                .insert(TilemapId(tilemap_entity))
+                .insert(LastUpdate::default())
+                .insert(TileVisible(true))
+                .id();
+            tile_storage.set(&tile_pos, Some(tile_entity));
         }
-        commands
-            .entity(tile_entity.unwrap())
-            .insert(LastUpdate::default());
-    });
+    }
```

Comment:
🙏🏻

To the completely ignorant of game dev person that I am, this is so much more intuitive. Can't wait for the post-rewrite version.